libitl: init at 0.1.8

amyipdev opened the pull request:
## Description of changes

This patch adds the Islamic Tools Library, which provides functions for calculating Islamic prayer times and Hijri calendar dates. It is necessary to add ITL before adding the itools suite (ical, ipraytime, ireminder, idate), which will be requested for addition to nixpkgs after libitl's acceptance.

Main repository: https://github.com/arabeyes-org/ITL
Homepage: https://www.arabeyes.org/ITL

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests)) N/A
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test) N/A
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages N/A
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [] Tested basic functionality of all binary files (usually in `./result/bin/`) N/A
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking N/A
  - [ ] (Module updates) Added a release notes entry if the change is significant N/A
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module N/A
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

azahi commented on pkgs/by-name/li/libitl/package.nix:
```diff
@@ -0,0 +1,39 @@
+    hash = "sha256-Hp+NXibSlF9LMRRlHWISAiQSz/dbMNjjEfKCNFVmIxo=";
+  };
+
+  buildPhase = ''
+    cmake .
+    make
+  '';
+
```
```suggestion
```
I believe this can be removed, defaualt buildPhase already executes make/cmake.

amyipdev replied:
Done.

azahi commented on pkgs/by-name/li/libitl/package.nix:
```diff
@@ -0,0 +1,39 @@
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib
+    mkdir -p $out/include/itl
+    cp libitl.so $out/lib/libitl.so.0
+    ln -s $out/lib/libitl.so.0 $out/lib/libitl.so
+    cp libitl.a $out/lib
+    cp $src/hijri/hijri.h $out/include/itl
+    cp $src/prayertime/prayer.h $out/include/itl
+  '';
```
Please reuse upstream CMake [install instructions](https://github.com/arabeyes-org/ITL/blob/master/CMakeLists.txt#L45-L48). These can be patched with `substituteInPlace` if needed.

amyipdev replied:
Done.

azahi commented on pkgs/by-name/li/libitl/package.nix:
```diff
@@ -0,0 +1,39 @@
+  '';
+
+  meta = with lib; {
+    description = "Islamic Tools and Libraries";
```
```suggestion
    description = "Islamic Tools and Libraries (ITL)";
    longDescription = ''
      The Islamic Tools and Libraries (ITL) project provides 
      a fully featured library for performing common Islamic calculations.
    '';
```

amyipdev replied:
Done. Not 100% on this one, but willing to take the suggestion and have implemented it.

azahi commented on pkgs/by-name/li/libitl/package.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchFromGitHub, cmake }:
```
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, cmake 
}:
```

amyipdev replied:
Is this necessary? Other packages use them in a single line.

azahi replied:
Single line diffs for subsequent PRs are harder to read. Multiline module inputs will be mandatory when the new formatter is going to be applied Nixpkgswide, albeit with slightly different syntax.

amyipdev replied:
@azahi I'll format it like this for now - but I will say that commas before instead of after look really, really terrible

azahi replied:
A matter of preference. This is notation is used pretty much everywhere in Nixpkgs. You can also use `nixfmt-rfc-style` to automatically format Nix code, this formatter has a slightly different approach for listing function arguments.

azahi commented on pkgs/by-name/li/libitl/package.nix:
```diff
@@ -0,0 +1,39 @@
+  version = "0.1.8";
+
+  nativeBuildInputs = [ cmake ];
+  outputs = [ "out" ];
```
```suggestion
```

This is implied by default.

amyipdev replied:
Done.

azahi reviewed (changes requested):
Welcome to Nixpkgs.

RaghavSood commented on pkgs/by-name/li/libitl/package.nix:
```diff
@@ -0,0 +1,39 @@
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libitl";
+  version = "0.1.8";
```
Can you clarify how this version number was selected? It doesn't seem to match any upstream tags or information.

If upstream does not have a reliable, maintained versioning scheme, and you are simply following the master branch (Which appears to be the case), then you should consider using the `unstable-<date>` format instead.

amyipdev replied:
The `CMakeLists.txt` has a version number, which is where I got that from.

amyipdev replied:
(And note, this also thus results in the main shared library output being `libitl.so.0.1.8`)

azahi replied:
Version should point to an upstream Git tag, if it's not present or if it's way too old, you can pin src to a specific commit, but then you'll have to set version as `VERSION-unstable-YYYY-MM-DD`, more on that [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming).

amyipdev commented:
PR is being moved back to draft status due to the merge of https://github.com/arabeyes-org/ITL/pull/53.

azahi commented:
Also, please use lowercase `init` in the commit message.

amyipdev commented:
@azahi will remake the commit with init lowercase - should documentation be updated to reflect that? I was going based on an example provided in documentation (I believe it was specifically with nginx)

amyipdev committed:
libitl: init at 0.1.8

This patch adds the Islamic Tools Library, which provides functions
for calculating Islamic prayer times and Hijri calendar dates. It is
necessary to add ITL before adding the itools suite (ical, ipraytime,
ireminder, idate), which will be requested for addition to nixpkgs
after libitl's acceptance.

Signed-off-by: Amy Parker <amy@amyip.net>

azahi commented:
> should documentation be updated to reflect that

It already [does](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions).
